addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1902,17 +1902,49 @@ var SnippetsMenu = Widget.extend({
             const $invisibleSnippets = globalSelector.all().find('.o_snippet_invisible').addBack('.o_snippet_invisible');
 
             $invisibleDOMPanelEl.toggleClass('d-none', !$invisibleSnippets.length);
-
-            const proms = _.map($invisibleSnippets, async el => {
-                const editor = await this._createSnippetEditor($(el));
+            // childrenPerSnippet -key: a snippet.
+            //                    -value: an array filled with its children
+            // snippets.
+            const childrenPerSnippet = new Map();
+            // Filter the $invisibleSnippets to only keep the non children
+            // snippets and create the map (childrenPerSnippet) of the snippets
+            // and their children.
+            const $rootInvisibleSnippets = $invisibleSnippets.filter((index, childSnippetEl) => {
+                for (const parentSnippetEl of $invisibleSnippets) {
+                    if (childSnippetEl !== parentSnippetEl
+                        && parentSnippetEl.contains(childSnippetEl)) {
+                        const childrenSnippets = childrenPerSnippet.get(parentSnippetEl) || [];
```

Comment:
I think that name it `childrenXXX` here is a bit confusing, the children here can be a HTMLElement wich is not a direct children but just a **descendant** right ?